common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -1009,7 +1009,8 @@ public static UTF8String concatWs(UTF8String separator, UTF8String... inputs) {
 
     // Allocate a new byte array, and copy the inputs one by one into it.
     // The size of the new array is the size of all inputs, plus the separators.
-    final byte[] result = new byte[numInputBytes + (numInputs - 1) * separator.numBytes];
+    int intNumInputBytes = Math.toIntExact(numInputBytes + (numInputs - 1) * separator.numBytes);
+    final byte[] result = new byte[intNumInputBytes];
```

Comment:
If we want to prevent overflow, we should do that for all operators:
```suggestion
    int resultSize = Math.addExact(
      numInputBytes,
      Math.multiplyExact(numInputs - 1, separator.numBytes));
    final byte[] result = new byte[resultSize];
```

Reply:
`numInputBytes` has  changed to long. you mean `(numInputs - 1) * separator.numBytes` can overflow ? yeah, maybe it can be. How about this ?
```
    int intNumInputBytes = Math.toIntExact(
            numInputBytes + (numInputs - 1) * (long)separator.numBytes);
```

Reply:
> you mean (numInputs - 1) * separator.numBytes can overflow ?

yep

> How about this ?

I am ok with that.

Reply:
updated